app/controllers/tasks.js

```diff
@@ -97,16 +97,23 @@ export default class TasksController extends Controller {
     const requestBody = { ...object };
     const taskCompletionPercentage = object.percentCompleted;
     if (taskCompletionPercentage) {
-      if (taskCompletionPercentage === TASK_PERCENTAGE.completedPercentage) {
-        this.isDevMode === true
-          ? (requestBody.status = 'DONE')
-          : (requestBody.status = 'COMPLETED');
+      if (
+        taskCompletionPercentage === TASK_PERCENTAGE.completedPercentage &&
+        !this.dev
+      ) {
+        requestBody.status = 'COMPLETED';
       }
       requestBody.percentCompleted = parseInt(taskCompletionPercentage);
     }
     return requestBody;
   }
-
+  getTaskById(taskId) {
+    const indexOfSelectedTask = this.allTasks.findIndex(
```

Comment:
what if indexOfSelectedTask === -1 ?

Reply:
It can not be -1 as that means there is task id which was not returned by the backend.(User is not involved in creating id)